--- a/usr/share/lipstick-jolla-home-qt5/layers/HomeLayer.qml
+++ b/usr/share/lipstick-jolla-home-qt5/layers/HomeLayer.qml
@@ -79,7 +79,7 @@ Pannable {
         height: rotation % 180 == 0
                     ? Lipstick.compositor.height
                     : Lipstick.compositor.width
-        z: -1000
+        z: currentItem != switcher ? -1000 : 100
 
         StatusBar {
             id: statusBar
@@ -89,7 +89,7 @@ Pannable {
             backgroundVisible: true
             y: -currentItem.statusOffset * (1.0-homescreen.progress) - (alternateItem ? alternateItem.statusOffset * homescreen.progress : 0)
             updatesEnabled: Lipstick.compositor.homeActive || Lipstick.compositor.homePeeking
-            opacity: Math.min(1.0, opacityFromY, currentItem.statusOpacity)
+            opacity: currentItem != switcher ? Math.min(1.0, opacityFromY, currentItem.statusOpacity) : 1.0
         }
     }
 
--- a/usr/share/lipstick-jolla-home-qt5/layers/SwitcherLayer.qml
+++ b/usr/share/lipstick-jolla-home-qt5/layers/SwitcherLayer.qml
@@ -10,7 +10,7 @@ PannableLayer {
     property alias overlayItem: overlay
 
     property real contentY
-    statusOffset: Math.min(contentY, statusBar.height + Theme.paddingMedium)
+    statusOffset: (desktop != null && desktop.switcher.housekeepingMenuActive) ? Math.min(contentY, statusBar.height + Theme.paddingMedium) : 0
 
     Item {
         id: content
--- a/usr/share/lipstick-jolla-home-qt5/statusarea/StatusBar.qml
+++ b/usr/share/lipstick-jolla-home-qt5/statusarea/StatusBar.qml
@@ -26,7 +26,15 @@ Item {
             bottomMargin: -Theme.paddingLarge
         }
         gradient: Gradient {
-            GradientStop { position: 0; color: Qt.rgba(0,0,0,0.1) }
+            GradientStop {
+                position: 0;
+                color: Theme.rgba(
+                    Theme.highlightDimmerColor,
+                    (desktop != null && desktop.switcher.housekeepingMenuActive) ?
+                        0.1 :
+                        0.7
+                )
+            }
             GradientStop { position: 1; color: "transparent" }
         }
         opacity: 1.0 - Math.abs(statusBar.y/Theme.paddingMedium)
--- a/usr/share/lipstick-jolla-home-qt5/switcher/Switcher.qml
+++ b/usr/share/lipstick-jolla-home-qt5/switcher/Switcher.qml
@@ -32,11 +32,6 @@ SilicaFlickable {
     onSwitcherVisibleChanged: {
         if (!switcherVisible) {
             closeApplicationEnabled = false
-            // The view is completely hidden. The delay is a grace period, so
-            // that if you quickly exit and reenter the view has not moved.
-            resetPosition(300)
-        } else {
-            cancelResetPosition()
         }
     }
 
@@ -346,7 +341,7 @@ SilicaFlickable {
             readonly property real minimumHeight: switcherRoot.height - statusBarHeight - ((largeScreen ? 5 : 3) * Theme.paddingLarge)
             readonly property real maximumWidth: switcherRoot.width - 2*Theme.paddingLarge
 
-            readonly property bool allowSmallCovers: !largeScreen
+            readonly property bool allowSmallCovers: false
             readonly property int smallColumns: (maximumWidth + spacing) / (Theme.coverSizeSmall.width + spacing)
 
             readonly property int largeColumns: Math.min(4, (maximumWidth + spacing) / (Theme.coverSizeLarge.width + spacing))
